### DISCUSSION TITLE
SQLite Auth

### DISCUSSION THREAD

**Discussion by rodydavis** [simolus3/drift#2540] [Ideas]
Not sure if this is being explored, but curious if it is possible to include the user auth logic from SQLite:

https://www.sqlite.org/src/doc/trunk/ext/userauth/user-auth.txt

**Comment by simolus3** [comment]
Most users probably don't need that feature, so I'm reluctant to add this to the native sqlite build we provide through `sqlite3_flutter_libs`. Exposing the API methods in the `sqlite3` package sounds fine to me, but I don't think we need support for them in `drift` (users can use the `setup` callback on `NativeDatabase` to call the authentication methods).

With the upcoming [native build support](https://github.com/dart-lang/native/) coming to Dart and Flutter, users would be able to select the compilation flags they want - so if we can make userauth opt-in, we should definitely support it. But in the current build model, managing different flavors is not possible on all platforms.

**Reply by rodydavis** [reply]
Sounds good! Just wanted to make an issue to track it if anyone was also looking for it.

**Reply by simolus3** [reply]
I've created a tracking issue on the repository for the lower-level sqlite3 bindings that would host these APIs :+1:

**Reply by rodydavis** [reply]
Thank you! What made me curious about it was seeing the go-sqlite library and the ability to use the methods:
https://github.com/mattn/go-sqlite3
